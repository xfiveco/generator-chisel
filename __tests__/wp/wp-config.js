const execa = require('execa');
const { answers: answersGenerator } = require('./helpers');

describe('WP wp-config command', () => {
  test('Test generating with wrong port first', async () => {
    const answers = answersGenerator();
    const answersLast = answers[answers.length - 1];
    answers.push({ ...answersLast });
    answersLast.databasePort = '3307';

    const consoleMock = jest.spyOn(console, 'log');

    await global.chiselTestHelpers.generateProjectWithAnswers(
      ['create', '--skip-format-and-build'],
      answers,
      { interceptWpConfig: true, mockRandomBytes: true },
    );

    global.chiselTestHelpers.normalizeConsoleMockCalls(consoleMock);
    expect(consoleMock.mock.calls).toMatchSnapshot();
  });

  test('Test generating with existing database', async () => {
    const answers = answersGenerator();
    const answersLast = answers[answers.length - 1];
    answers.push({ useExisting: false });

    await execa('mysql', [
      '-e',
      `CREATE DATABASE \`${answersLast.databaseName}\``,
      '-u',
      'root',
    ]);

    await global.chiselTestHelpers.generateProjectWithAnswers(
      ['create', '--skip-format-and-build'],
      answers,
      { interceptWpConfig: true, mockRandomBytes: true },
    );
  });
});
