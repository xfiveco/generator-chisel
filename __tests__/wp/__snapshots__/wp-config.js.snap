// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WP wp-config command Test generating with wrong port first 1`] = `
Array [
  Array [
    "",
  ],
  Array [
    "[31;1mError:[0m Failed to get current SQL modes. Reason: ERROR 2003 (HY000): Can't connect to MySQL server on '127.0.0.1' (--ERROR-DETAILS-)
",
  ],
  Array [
    "",
  ],
  Array [
    "Trying again...",
  ],
  Array [
    "",
  ],
]
`;
